go/service/chat_local.go

```diff
@@ -219,6 +219,9 @@ func (h *chatLocalHandler) GetThreadLocal(ctx context.Context, arg chat1.GetThre
 		return chat1.GetThreadLocalRes{}, err
 	}
 
+	// Run type filter if it exists
+	thread.Messages = chat.FilterByType(thread.Messages, arg.Query)
```

Comment:
If we make this filter accept boxed messages, and move this to before `unboxThread`, I think we can just use the message type from the server header.

(Or did you want to use unboxed version to make it less prone to server abuse?)